dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -439,9 +439,12 @@ escapeLabel :: Bool -> Text -> Text
 escapeLabel allowReserved l =
     case Text.uncons l of
         Just (h, t)
-            | headCharacter h && Text.all tailCharacter t && (allowReserved || not (Data.HashSet.member l reservedIdentifiers))
+            | headCharacter h && Text.all tailCharacter t && someOrNotLanguageKeyword && (allowReserved || notReservedIdentifier)
```

Comment:
You can make this check more efficient by refactoring it to:

```haskell
| headCharacter h
    &&  Text.all tailCharacter t
    &&  (   not (member l reservedIdentifiers)
        ||  (allowReserved && (not (member l reservedKeywordsExceptSome)))
        )
```

... that way the vast majority of the time `escapeLabel` function doesn't need to check whether `allowReserved` is `True` or not

Reply:
In some moment we may need a refactor for special handling of `Some`, in case that in the future another keyword requires that special treatment as well.